mne/viz/_figure.py

```diff
@@ -0,0 +1,1854 @@
+        super().__init__(figsize=figsize)
+        # add our param object
+        self.mne = MNEFigParams(**kwargs)
+
+    def _close(self, event):
+        """Handle close events."""
+        # remove references from parent fig to child fig
+        is_child = getattr(self.mne, 'parent_fig', None) is not None
+        is_named = getattr(self.mne, 'fig_name', None) is not None
+        if is_child:
+            self.mne.parent_fig.mne.child_figs.remove(self)
+            if is_named:
+                setattr(self.mne.parent_fig.mne, self.mne.fig_name, None)
+
+    def _keypress(self, event):
+        """Handle keypress events."""
+        if event.key == self.mne.close_key:
+            from matplotlib.pyplot import close
+            close(self)
+        elif event.key == 'f11':  # full screen
+            self.canvas.manager.full_screen_toggle()
+
+    def _buttonpress(self, event):
+        """Handle buttonpress events."""
+        pass
+
+    def _pick(self, event):
+        """Handle matplotlib pick events."""
+        pass
+
+    def _resize(self, event):
+        """Handle window resize events."""
+        pass
+
+    def _add_default_callbacks(self, **kwargs):
+        """Remove some matplotlib default callbacks and add MNE-Python ones."""
+        # Remove matplotlib default keypress catchers
+        default_callbacks = list(
+            self.canvas.callbacks.callbacks.get('key_press_event', {}))
+        for callback in default_callbacks:
+            self.canvas.callbacks.disconnect(callback)
+        # add our event callbacks
+        callbacks = dict(resize_event=self._resize,
+                         key_press_event=self._keypress,
+                         button_press_event=self._buttonpress,
+                         close_event=self._close,
+                         pick_event=self._pick)
+        callbacks.update(kwargs)
+        callback_ids = dict()
+        for event, callback in callbacks.items():
+            callback_ids[event] = self.canvas.mpl_connect(event, callback)
+        # store callback references so they aren't garbage-collected
+        self.mne._callback_ids = callback_ids
+
+    def _get_dpi_ratio(self):
+        """Get DPI ratio (to handle hi-DPI screens)."""
+        dpi_ratio = 1.
+        for key in ('_dpi_ratio', '_device_scale'):
+            dpi_ratio = getattr(self.canvas, key, dpi_ratio)
+        return dpi_ratio
+
+    def _get_size_px(self):
+        """Get figure size in pixels."""
+        dpi_ratio = self._get_dpi_ratio()
+        return self.get_size_inches() * self.dpi / dpi_ratio
+
+    def _inch_to_rel(self, dim_inches, horiz=True):
+        """Convert inches to figure-relative distances."""
+        fig_w, fig_h = self.get_size_inches()
+        w_or_h = fig_w if horiz else fig_h
+        return dim_inches / w_or_h
+
+
+class MNEAnnotationFigure(MNEFigure):
+    """Interactive dialog figure for annotations."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
```

Comment:
```suggestion
```
I don't think these are needed. AFAICT all it does is ensure all are `kwargs` but `MNEFigure` already ensures this.